File: 10 - Palindrome-Check/10.js
// recursive solution:

function isPalindromeRec(string){
    if (string.length <= 1){
        return true
    }
    if (string[0] === string.at(-1)){
        return isPalindromeRec(string.slice(1,-1))
}
    else{
        return false
    }
}

// iterative solution

function isPalindrome(string){
    while (string.length > 1){
        if (string[0] === string.at(-1)){
            string = string.slice(1,-1)
        }
        else{
            return false
        }
    }
    return true
}

File: 10 - Palindrome-Check/10.py
# recursive solution:


def is_palindrome_rec(string):
    if len(string) <= 1:
        return True
    if string[0] == string[-1]:
        return is_palindrome_rec(string[1:-1])
    else:
        return False


# iterative solution


def is_palindrome(string):
    while len(string) > 1:
        if string[0] == string[-1]:
            string = string[1:-1]
        else:
            return False
    return True
